### PR TITLE
refactor: purging constant `MAX_GAS_PER_TX`

### PR DESCRIPTION
**Pull request by Torres-ssf**
In our ongoing effort to eliminate the hardcoded values and fetch them from the chain, I am purging the `MAX_GAS_PER_TX` constant. 

This change aligns with the changes made on #1271

### CODE REVIEW

File: packages/transactions/src/configs.ts [hunk 1]
```diff
@@ -6,9 +6,6 @@ export const CONTRACT_MAX_SIZE = 16 * 1024;
 /** Maximum number of witnesses. */
 export const MAX_WITNESSES = 16;
 
-/** Maximum gas per transaction. */
-export const MAX_GAS_PER_TX = bn(10_000_000);
```
Comment (LuizAsFight) [inline packages/transactions/src/configs.ts:10]: remove import { bn } also

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

File: packages/transactions/src/configs.ts (hunk 1 above)
Comment (Dhaiwat10) [inline packages/transactions/src/configs.ts:10]: Looks like you purged the constant `MAX_GAS_PER_TX` but the PR name and the changeset mentions `MAX_GAS_PER_PREDICATE` instead

Comment (Torres-ssf) [reply packages/transactions/src/configs.ts:10]: You're right :man_facepalming: 

Fixed

**Review by arboleya — DISMISSED** [review]
This case should not yield anything like that, but, in general, I believe that everything test-related should be inside the `describe` block to avoid leakage and weird behaviors resulting from having code executed ahead of time before the tests are run.

Also unrelated to your changes, but I wonder if it wouldn't be more straightforward to make the `accountOrProvider` parameter mandatory on the constructor, which could make the provider validations redundant.

I may be missing some use cases.

File: packages/contract/src/contract-factory.ts [hunk 2]
```diff
@@ -109,11 +108,19 @@ export default class ContractFactory {
       storageSlots: storageSlots || [],
     };
 
+    if (!this.provider) {
+      throw new FuelError(
+        ErrorCode.MISSING_PROVIDER,
+        'Cannot create transaction request without provider'
+      );
+    }
```
[the remark this answers is not in the document]
Comment (Torres-ssf) [reply packages/contract/src/contract-factory.ts]: @camsjams I think that makes sense. 

But that implies that we would make the param `accountOrProvider` required in the class constructor. Then this validation won't be required:
https://github.com/FuelLabs/fuels-ts/blob/995401aafb8489c531d9b868c51bac04872fc1bb/packages/contract/src/contract-factory.ts#L42-L76

But I wonder why it was possible to instantiate a `ContractFactory` without a provider.

**Review by arboleya — DISMISSED** [review]

**Review by Dhaiwat10 — APPROVED** [review]

File: packages/contract/src/contract-factory.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (arboleya) [reply packages/contract/src/contract-factory.ts:112]: @camsjams Why can't we make the `accountOrProvider` parameter required in the constructor, remove the `connect` method, and the provider validation? The `connect` method accepts `null` as a Provider, which makes no sense. 🤔 

Something seems off.

**Review by digorithm — APPROVED** [review]

File: packages/contract/src/contract-factory.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (digorithm) [reply packages/contract/src/contract-factory.ts:112]: Agreed -- should this conversation be resolved?

**Review by LuizAsFight — APPROVED** [review]

**Pull request merged**